Chore/ddw 218 Improve request body params length calculation

nikolaglumac opened the pull request:
This PR improves the request body length calculation as per recommendation by @cleverca22 given in: https://github.com/input-output-hk/daedalus/pull/839#issuecomment-378896197.

KtorZ commented:
Where does that `Buffer` comes from in browsers ^.^ ?

nikolaglumac commented:
@KtorZ in this case it comes from node. Daedalus is in the end Electron app so we have this at our disposal ;)

DominikGuzei commented:
@nikolaglumac @KtorZ please be aware though that this will soon change since we will disable node integration in this sprint … so it would be better if you find a solution without node integration before we merge this change 😉

DominikGuzei commented:
@nikolaglumac i actually forgot what the initial reason was for the change? If it was performance we can safely abort this work since this is not a performance critical part of the application.

nikolaglumac commented:
@DominikGuzei yes the performance was the reason.

nikolaglumac commented:
@DominikGuzei @KtorZ @cleverca22 there is apparently a "browser" version of node.js Buffer: https://github.com/feross/buffer
Maybe we could use this one? My question is if this would really be an advantage over the current implementation with `(new TextEncoder()).encode(requestBody).length`.
I would like to hear your opinion on this one...

KtorZ commented:
In JavaScript (or any language in general), I'd rather rely on built-in libraries than pulling in an extra dependency. 
I am not sure that the "performance" argument has to come into play in that as we're probably comparing nanoseconds here. The Buffer interface is definitely more elegant, though, if we move the instantiation of the `TextEncoder` elsewhere, we would also have something quite neat:

```
encoder.encode(requestBody).length
```

cleverca22 commented on source/renderer/app/api/ada/lib/request.js:
```diff
@@ -54,7 +54,7 @@ function typedRequest<Response>(
       requestBody = JSON.stringify(rawBodyParams);
       options.headers = {
-        'Content-Length': (new TextEncoder()).encode(requestBody).length,
+        'Content-Length': Buffer.byteLength(requestBody),
```
i'm thinking do a full `var bytes = Buffer(requestBody,"utf-8");` and then send that exact buffer when doing the post, so the post cant use a different encoding

nikolaglumac commented:
@KtorZ I like your proposal 👍 - I guess that is the best thing we can do at the moment since we will remove node integration from the renderer process...

nikolaglumac commented:
@DominikGuzei @KtorZ @cleverca22 I have now reverted the code to the old `TextEncoder` solution with the difference that I have extracted this part into a util which is then reused in all of the 3 request files...

nikolaglumac commented:
@KtorZ in this case I will leave it as is since it is used only for the requests and thus it is placed in api/lib/utils. In case we will need it elsewhere we will rename it and move it into a more global directory.
